app/controllers/coronavirus_form/product_details_controller.rb

```diff
@@ -37,43 +38,45 @@ def destroy
   helper_method :selected_ppe?
 
   def selected_ppe?
-    @product["medical_equipment_type"] == I18n.t(
+    @product[:medical_equipment_type] == I18n.t(
       "coronavirus_form.questions.medical_equipment_type.options.number_ppe.label",
     )
   end
 
   def selected_made_in_uk?
-    @product["product_location"] == I18n.t(
+    @product[:product_location] == I18n.t(
       "coronavirus_form.questions.product_details.product_location.options.option_uk.label",
     )
   end
 
-  def validate_fields(product)
-    missing_fields = validate_missing_fields(product)
-    product_equipment_type = selected_ppe? ? validate_radio_field("#{controller_name}.equipment_type", radio: product["equipment_type"]) : []
-    product_location_validation = validate_radio_field("#{controller_name}.product_location", radio: product["product_location"])
-    postcode_validation = validate_product_postcode(product)
-    missing_fields + product_equipment_type + product_location_validation + postcode_validation
+  def validate_fields
+    [
+      validate_field_response_length(controller_name, TEXT_FIELDS),
+      validate_missing_fields(@product),
+      selected_ppe? ? validate_radio_field("#{controller_name}.equipment_type", radio: @product[:equipment_type]) : [],
```

Comment:
This could just be an inline if:
```
validate_radio_field("#{controller_name}.equipment_type", radio: @product[:equipment_type]) if selected_ppe?
```
the `nil` returned by the conditional being false will be removed by the `compact` anyway.
```

Reply:
Unfortunately the linter doesn't permit inline if statements in an array.